python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -970,6 +993,42 @@ cdef class Column:
 
         flat, shape = _flatten_nested_list(obj, depth)
 
+        if dtype.id() == type_id.STRING:
+            encoded = [s.encode("utf-8") for s in flat]
+            offsets = [0] + list(accumulate(len(s) for s in encoded))
+            joined_bytes = b"".join(encoded)
+
+            offsets_col = Column.from_iterable_of_py(
+                offsets, dtype=DataType(type_id.INT32)
+            )
+
+            mv = memoryview(array.array("B", joined_bytes))
+
+            iface = {
+                "data": (mv.obj.buffer_info()[0], False),
+                "shape": (len(joined_bytes),),
+                "typestr": "|u1",
+                "strides": None,
+                "version": 3,
+            }
+
+            chars_col = Column.from_array_interface(ArrayInterfaceWrapper(iface))
+
+            base = Column(
+                DataType(type_id.STRING),
+                len(flat),
+                chars_col.data(),
+                None,
+                0,
+                0,
+                [offsets_col, chars_col],
```

Comment:
libcudf strings columns only have one child column
The chars data is a device-buffer.